### PR TITLE
Update sprints() calls to snprintf() to silence compiler warnings 

### PR DESCRIPTION
**Pull request by Tiza59**

### CODE REVIEW

File: solver.cc [hunk 1]
```diff
@@ -2054,7 +2054,7 @@ class InteractivePlay {
 
   char* ColoredNameOf(int card) {
     static char name[32];
-    snprintf(name, "%s %c", SuitSign(SuitOf(card)), NameOf(card)[1]);
+    snprintf(name, 32, "%s %c", SuitSign(SuitOf(card)), NameOf(card)[1]);
```
Comment (macroxue) [inline solver.cc]: Please change `32` to `sizeof(name)`.

File: solver.cc [hunk 2]
```diff
@@ -1876,13 +1876,13 @@ class InteractivePlay {
       ns_contract = true;
       int level = (TOTAL_TRICKS - num_tricks) + target_ns_tricks - 6;
       auto declarer = starting_ns_tricks == 0 ? SeatName((lead_seat + 3) % 4) : "NS";
-      snprintf(contract, "%d%s by %s", level, SuitSign(trump), declarer);
+      snprintf(contract, 80, "%d%s by %s", level, SuitSign(trump), declarer);
```
Comment (macroxue) [inline solver.cc]: Please change `80` to `sizeof(contract)`, so we don't have to modify this code again when the size of the `contract` array changes. Same for the next occurrence.

**Review by macroxue — CHANGES_REQUESTED** [review]
Thanks for submitting this patch. Please make minor modifications as suggested in the file.

**Comment by Tiza59** [comment]
Thank you for helping me to improve this little change.

**Pull request closed without merging**